### ISSUE TITLE
Encryption Recovery Key not working in 9.1.4

### ISSUE THREAD

**Issue by tervor** [owncloud/core#27547]
I encounter an issue with the recovery key and resetting a users password from the admin account. I have the default encryption module in place and a recovery key set.
Then I create a user, go tho his preferences and enable the recovery key function. Then I login to the admin user again and reset the users password. I do this by entering the recovery key in the field on top of the users view and then entering the password at the designated user. I am now able to login with the user using the new password, but I immediately get the following message on top of the page: 
> Invalid private key for Encryption App. Please update your private key password in your 
personal settings to recover access to your encrypted files.

If I try to access the the files in the users account I get this error:
> An error occurred!
>
>Private Key missing for user: please try to log-out and log-in again

I am now able to change the recovery Key of the user in the users preferences.
This happens repetitive.  

Taking a look to the log shows the following Exception (passwords censored):
```
{"reqId":"yRM6fEvUcpsBjd6WsmiN","remoteAddr":"***Censored***","app":"no app in context","message":"Exception: {\"Exception\":\"OCA\\\\Encryption\\\\Exceptions\\\\PrivateKeyMissingException\",\"Message\":\"Private Key missing for user: please try to log-out and log-in again\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/Hooks\\\/UserHooks.php(230): OCA\\\\Encryption\\\\Session->getPrivateKey()\\n#1 [internal function]: OCA\\\\Encryption\\\\Hooks\\\\UserHooks->setPassphrase(Array)\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/hook.php(105): call_user_func(Array, Array)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Server.php(265): OC_Hook::emit('OC_User', 'post_setPasswor...', Array)\\n#4 [internal function]: OC\\\\Server->OC\\\\{closure}(Object(OC\\\\User\\\\User), '***new user password***', '***recovery key***')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/EmitterTrait.php(98): call_user_func_array(Object(Closure), Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/PublicEmitter.php(32): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postSetPassword', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/User\\\/User.php(236): OC\\\\Hooks\\\\PublicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postSetPassword', Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/user.php(385): OC\\\\User\\\\User->setPassword('***new user password***', '***recovery key***')\\n#9 \\\/var\\\/www\\\/owncloud\\\/settings\\\/ChangePassword\\\/Controller.php(138): OC_User::setPassword('kevin', '***new user password***', '***recovery key***')\\n#10 [internal function]: OC\\\\Settings\\\\ChangePassword\\\\Controller::changeUserPassword(Array)\\n#11 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(280): call_user_func(Array, Array)\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(891): OC\\\\Route\\\\Router->match('\\\/settings\\\/users...')\\n#13 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/Session.php\",\"Line\":78}","level":3,"time":"2017-03-30T14:54:25+00:00","method":"POST","url":"\/index.php\/settings\/users\/changepassword","user":"admin"}
 ```
Note: this was performed on our testing ownCloud installation.

### Steps to reproduce
1. Enable the default encryption module
2. Set a recovery Key
3. Create a user
4. Login as this user
5. Enable the recovery key
6. Login back as the admin
7. Reset the password of the user (fill in the recovery key)
8. Check the result by logging in as the user with the new password.

### Expected behaviour
The users key should have been updated. The user does not need to take any action after login with the new password in order to access his files.

### Actual behaviour
The users key does not get updated. The user has to update his key using his old password and new one. In a case of a forgotten password, the user does not know the old password and looses access to all his data!

### Server configuration
**Operating system**: Debian Jessie 8.7

**Web server:** Apache/2.4.10

**Database:** mysql  Ver 14.14 Distrib 5.5.54

**PHP version:** PHP 5.6.30-0+deb8u1

**ownCloud version:** 9.1.4.2

**Updated from an older ownCloud or fresh install:** Updated from 9.1.0

**Where did you install ownCloud from:** Download from owncloud.org

**Signing status (ownCloud 9.0 and above):**

```
Integrity checker has been disabled. Integrity cannot be verified.
```
After I enabled the check again and run it with occ integrity:check-core this result occurs:
```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.3.2
  - comments: 0.3.0
  - configreport: 0.1.1
  - dav: 0.2.7
  - encryption: 1.3.0
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
Disabled:
  - documents
  - external
  - files_antivirus
  - files_external
  - password_policy
  - user_external
  - user_ldap
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "ocheliky9brf",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "\/var\/www\/owncloud-data",
        "overwrite.cli.url": "http:\/\/***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "9.1.4.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "session_lifetime": 60,
        "theme": "ctu",
        "singleuser": false,
        "maintenance": false,
        "loglevel": 0
    },
    "apps": {
        "activity": {
            "enabled": "yes",
            "installed_version": "2.3.2",
            "types": "filesystem"
        },
        "backgroundjob": {
            "lastjob": "4"
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": "logging"
        },
        "configreport": {
            "enabled": "yes",
            "installed_version": "0.1.1",
            "types": "filesystem"
        },
        "core": {
            "default_encryption_module": "OC_DEFAULT_MODULE",
            "encryption_enabled": "yes",
            "installedat": "1474293063.3629",
            "lastcron": "1490945580",
            "lastupdateResult": "[]",
            "lastupdatedat": "1490945257",
            "oc.integritycheck.checker": "[]",
            "public_documents": "documents\/public.php",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "umgmt_send_email": "false",
            "umgmt_show_backend": "false",
            "umgmt_show_email": "true",
            "umgmt_show_last_login": "false",
            "umgmt_show_storage_location": "false",
            "updater.secret.created": "1490025303"
        },
        "dav": {
            "enabled": "yes",
            "installed_version": "0.2.7",
            "types": "filesystem"
        },
        "documents": {
            "enabled": "no",
            "installed_version": "0.13.1",
            "ocsid": "168711",
            "types": ""
        },
        "encryption": {
            "enabled": "yes",
            "encryptHomeStorage": "1",
            "installed_version": "1.3.0",
            "masterKeyId": "master_41aa90db",
            "publicShareKeyId": "pubShare_41aa90db",
            "recoveryAdminEnabled": "1",
            "recoveryKeyId": "recoveryKey_41aa90db",
            "types": "filesystem"
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": ""
        },
        "federation": {
            "enabled": "yes",
            "installed_version": "0.1.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "default_quota": "5 GB",
            "enabled": "yes",
            "installed_version": "1.5.1",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "0.8.1",
            "ocsid": "166049",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "installed_version": "0.10.0",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "yes",
            "installed_version": "2.1",
            "ocsid": "166051",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "0.9.0",
            "types": "filesystem"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": "filesystem"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "0.9.8",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "yes",
            "installed_version": "1.1",
            "ocsid": "166055",
            "types": ""
        },
        "gallery": {
            "enabled": "yes",
            "installed_version": "15.0.0",
            "types": ""
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": "logging"
        },
        "password_policy": {
            "enabled": "no",
            "installed_version": "0.2",
            "min_length": "8",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "0.5.0",
            "types": "prevent_group_restriction"
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": "logging"
        },
        "templateeditor": {
            "enabled": "yes",
            "installed_version": "0.1",
            "types": ""
        },
        "updatenotification": {
            "core": "9.1.4",
            "enabled": "yes",
            "installed_version": "0.2.1",
            "types": ""
        }
    }
}
```

**Are you using external storage, if yes which one:** No, local storage only

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** No, only local

### Client configuration
**Browser:** Firefox 52.0.1 (64-bit)

**Operating system:** Mac OS X 10.9.5

#### Web server error log
I censored some not related entries like access to a not existing file (not related to owncloud) or not related hostname.
```
[Thu Mar 30 06:25:01.918104 2017] [ssl:warn] [pid 11567] AH01909: ***Censored***:443:0 server certificate does NOT include an ID which matches the server name
[Thu Mar 30 06:25:01.918286 2017] [mpm_prefork:notice] [pid 11567] AH00163: Apache/2.4.10 (Debian) OpenSSL/1.0.1t configured -- resuming normal operations
[Thu Mar 30 06:25:01.918293 2017] [core:notice] [pid 11567] AH00094: Command line: '/usr/sbin/apache2'
[Thu Mar 30 09:57:48.880032 2017] [:error] [pid 30464] [client ***Censored***:6648] script '/var/www/html/azenv.php' not found or unable to stat, referer: ***Censored***
[Thu Mar 30 14:12:42.558853 2017] [:error] [pid 30460] [client ***Censored***:30507] script '/var/www/html/azenv.php' not found or unable to stat, referer: ***Censored***
[Thu Mar 30 18:34:42.492224 2017] [:error] [pid 3787] [client ***Censored***:60029] script '/var/www/html/azenv.php' not found or unable to stat, referer: ***Censored***
[Thu Mar 30 21:34:06.811036 2017] [ssl:error] [pid 3494] [client ***Censored***:36820] AH02042: rejecting client initiated renegotiation
[Thu Mar 30 22:57:24.061370 2017] [:error] [pid 3629] [client ***Censored***:18175] script '/var/www/html/azenv.php' not found or unable to stat, referer: ***Censored***
[Fri Mar 31 03:05:37.805159 2017] [:error] [pid 3787] [client ***Censored***:31138] script '/var/www/html/azenv.php' not found or unable to stat, referer: ***Censored***
[Fri Mar 31 06:25:02.854357 2017] [mpm_prefork:notice] [pid 11567] AH00171: Graceful restart requested, doing restart
```

#### ownCloud log (data/owncloud.log)
I also censored here some ip addresses. The log contains entires from multiple test runs. Also I had to censor some passwords that were printed as clear text in the logs!
```
{"reqId":"4JQQYYsDQa\/P5J8ZpKo7","remoteAddr":"***Censored***","app":"core","message":"No update found at the ownCloud appstore for app 166055","level":0,"time":"2017-03-30T14:46:53+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=enabled&includeUpdateInfo=1","user":"admin"}
{"reqId":"4JQQYYsDQa\/P5J8ZpKo7","remoteAddr":"***Censored***","app":"core","message":"No update found at the ownCloud appstore for app 166051","level":0,"time":"2017-03-30T14:46:54+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=enabled&includeUpdateInfo=1","user":"admin"}
{"reqId":"4JQQYYsDQa\/P5J8ZpKo7","remoteAddr":"***Censored***","app":"core","message":"No update found at the ownCloud appstore for app 166049","level":0,"time":"2017-03-30T14:46:54+00:00","method":"GET","url":"\/index.php\/settings\/apps\/list?category=enabled&includeUpdateInfo=1","user":"admin"}
{"reqId":"FvDmMP4Y\/3fLh6Sal8Ea","remoteAddr":"***Censored***","app":"PHP","message":"Non-static method OCA\\PasswordPolicyEnforcement\\Hooks\\Hooks::preSetPassword() should not be called statically at \/var\/www\/owncloud\/lib\/private\/Hooks\/EmitterTrait.php#36","level":3,"time":"2017-03-30T14:46:58+00:00","method":"POST","url":"\/index.php\/settings\/ajax\/disableapp.php","user":"admin"}
{"reqId":"sG3UXXgQX\/0CZJt\/ZD0H","remoteAddr":"***Censored***","app":"DeleteOrphanedSharesJob","message":"0 orphaned share(s) deleted","level":0,"time":"2017-03-30T14:47:14+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"T7KvnV4zKJVHHnzq44oG","remoteAddr":"***Censored***","app":"core","message":"Login failed: 'kecin' (Remote IP: '***Censored***')","level":2,"time":"2017-03-30T14:47:27+00:00","method":"POST","url":"\/index.php\/login","user":"--"}
{"reqId":"Put0OSt6EMpSa1li1l\/g","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"kevin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T14:47:35+00:00","method":"POST","url":"\/index.php\/login?user=kecin","user":"kevin"}
{"reqId":"rqpJFzcecYTBa1h1Ewg+","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"admin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T14:48:51+00:00","method":"POST","url":"\/index.php\/login","user":"admin"}
{"reqId":"gKgTMJMYceI51zqvaFdy","remoteAddr":"***Censored***","app":"no app in context","message":"Exception: {\"Exception\":\"OCA\\\\Encryption\\\\Exceptions\\\\PrivateKeyMissingException\",\"Message\":\"Private Key missing for user: please try to log-out and log-in again\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/Hooks\\\/UserHooks.php(230): OCA\\\\Encryption\\\\Session->getPrivateKey()\\n#1 [internal function]: OCA\\\\Encryption\\\\Hooks\\\\UserHooks->setPassphrase(Array)\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/hook.php(105): call_user_func(Array, Array)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Server.php(265): OC_Hook::emit('OC_User', 'post_setPasswor...', Array)\\n#4 [internal function]: OC\\\\Server->OC\\\\{closure}(Object(OC\\\\User\\\\User), '***new user password***', '***recovery key***')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/EmitterTrait.php(98): call_user_func_array(Object(Closure), Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/PublicEmitter.php(32): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postSetPassword', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/User\\\/User.php(236): OC\\\\Hooks\\\\PublicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postSetPassword', Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/user.php(385): OC\\\\User\\\\User->setPassword('***new user password***', '***recovery key***')\\n#9 \\\/var\\\/www\\\/owncloud\\\/settings\\\/ChangePassword\\\/Controller.php(138): OC_User::setPassword('kevin', '***new user password***', '***recovery key***')\\n#10 [internal function]: OC\\\\Settings\\\\ChangePassword\\\\Controller::changeUserPassword(Array)\\n#11 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(280): call_user_func(Array, Array)\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(891): OC\\\\Route\\\\Router->match('\\\/settings\\\/users...')\\n#13 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/Session.php\",\"Line\":78}","level":3,"time":"2017-03-30T14:49:43+00:00","method":"POST","url":"\/index.php\/settings\/users\/changepassword","user":"admin"}
{"reqId":"\/xAhJ7+69oeraOHELY79","remoteAddr":"***Censored***","app":"no app in context","message":"Invalidating tokens older than 2017-03-30T14:48:48+00:00","level":1,"time":"2017-03-30T14:49:48+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"hA000WbgC6qi3zlfYPDS","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"kevin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T14:49:55+00:00","method":"POST","url":"\/index.php\/login","user":"kevin"}
{"reqId":"9kzWQ26kPz7+64I\/J3e9","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"kevin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T14:50:38+00:00","method":"POST","url":"\/index.php\/login","user":"kevin"}
{"reqId":"1oj18A3BrKJKRPnwDer0","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"admin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T14:52:08+00:00","method":"POST","url":"\/index.php\/login","user":"admin"}
{"reqId":"\/4HU7MzWc5klevJEt8aN","remoteAddr":"***Censored***","app":"no app in context","message":"Invalidating tokens older than 2017-03-30T14:52:44+00:00","level":1,"time":"2017-03-30T14:53:44+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"a+7A8y8XBO631oLOHE4h","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"admin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T14:53:49+00:00","method":"POST","url":"\/index.php\/login","user":"admin"}
{"reqId":"yRM6fEvUcpsBjd6WsmiN","remoteAddr":"***Censored***","app":"no app in context","message":"Exception: {\"Exception\":\"OCA\\\\Encryption\\\\Exceptions\\\\PrivateKeyMissingException\",\"Message\":\"Private Key missing for user: please try to log-out and log-in again\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/Hooks\\\/UserHooks.php(230): OCA\\\\Encryption\\\\Session->getPrivateKey()\\n#1 [internal function]: OCA\\\\Encryption\\\\Hooks\\\\UserHooks->setPassphrase(Array)\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/hook.php(105): call_user_func(Array, Array)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Server.php(265): OC_Hook::emit('OC_User', 'post_setPasswor...', Array)\\n#4 [internal function]: OC\\\\Server->OC\\\\{closure}(Object(OC\\\\User\\\\User), '***new user password***', '***recovery key***')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/EmitterTrait.php(98): call_user_func_array(Object(Closure), Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Hooks\\\/PublicEmitter.php(32): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postSetPassword', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/User\\\/User.php(236): OC\\\\Hooks\\\\PublicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postSetPassword', Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/user.php(385): OC\\\\User\\\\User->setPassword('***new user password***', '***recovery key***')\\n#9 \\\/var\\\/www\\\/owncloud\\\/settings\\\/ChangePassword\\\/Controller.php(138): OC_User::setPassword('kevin', '***new user password***', '***recovery key***')\\n#10 [internal function]: OC\\\\Settings\\\\ChangePassword\\\\Controller::changeUserPassword(Array)\\n#11 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(280): call_user_func(Array, Array)\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(891): OC\\\\Route\\\\Router->match('\\\/settings\\\/users...')\\n#13 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/Session.php\",\"Line\":78}","level":3,"time":"2017-03-30T14:54:25+00:00","method":"POST","url":"\/index.php\/settings\/users\/changepassword","user":"admin"}
{"reqId":"hrRET1d+xYq+\/DOLMH9Y","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"kevin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T14:56:54+00:00","method":"POST","url":"\/index.php\/login","user":"kevin"}
{"reqId":"kprPZcsYh2QZqpQ9zjnj","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"kevin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T15:05:05+00:00","method":"POST","url":"\/index.php\/login","user":"kevin"}
{"reqId":"QJcaIe4FhY+2qe6OaEL3","remoteAddr":"***Censored***","app":"core","message":"No update found at the ownCloud appstore for app 166049","level":0,"time":"2017-03-30T15:05:07+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"QJcaIe4FhY+2qe6OaEL3","remoteAddr":"***Censored***","app":"core","message":"No update found at the ownCloud appstore for app 166051","level":0,"time":"2017-03-30T15:05:07+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"QJcaIe4FhY+2qe6OaEL3","remoteAddr":"***Censored***","app":"core","message":"No update found at the ownCloud appstore for app 166055","level":0,"time":"2017-03-30T15:05:08+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"3O19L8+joYreJGnIW8H9","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"kevin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-30T16:18:01+00:00","method":"GET","url":"\/ocs\/v2.php\/apps\/notifications\/api\/v1\/notifications?format=json","user":"kevin"}
{"reqId":"PIiAVyGLYkSBhXpTklQW","remoteAddr":"***Censored***","app":"DeleteOrphanedItems","message":"0 orphaned system tag relations deleted","level":0,"time":"2017-03-31T07:25:08+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"PIiAVyGLYkSBhXpTklQW","remoteAddr":"***Censored***","app":"DeleteOrphanedItems","message":"0 orphaned user tag relations deleted","level":0,"time":"2017-03-31T07:25:08+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"PIiAVyGLYkSBhXpTklQW","remoteAddr":"***Censored***","app":"DeleteOrphanedItems","message":"0 orphaned comments deleted","level":0,"time":"2017-03-31T07:25:08+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"PIiAVyGLYkSBhXpTklQW","remoteAddr":"***Censored***","app":"DeleteOrphanedItems","message":"0 orphaned comment read marks deleted","level":0,"time":"2017-03-31T07:25:08+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"+HCdhJLTmupasL2xCQEQ","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"admin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-31T07:27:37+00:00","method":"POST","url":"\/index.php\/login","user":"admin"}
{"reqId":"ivMknO8QTlgOP4YyO4p6","remoteAddr":"***Censored***","app":"no app in context","message":"Invalidating tokens older than 2017-03-31T07:26:38+00:00","level":1,"time":"2017-03-31T07:27:38+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"d571dp2Xmkr2Rr9z4e7a","remoteAddr":"***Censored***","app":"no app in context","message":"Current user is not logged in","level":0,"time":"2017-03-31T07:32:41+00:00","method":"GET","url":"\/index.php\/settings\/integrity\/failed","user":"--"}
{"reqId":"h+\/wgxdr\/Lvdi9tHBlf5","remoteAddr":"***Censored***","app":"no app in context","message":"Could not decrypt the private key from user \"admin\"\" during login. Assume password change on the user back-end. Error message: Bad Signature","level":2,"time":"2017-03-31T07:32:47+00:00","method":"POST","url":"\/index.php\/login?redirect_url=%252Findex.php%252Fsettings%252Fintegrity%252Ffailed","user":"admin"}
{"reqId":"uUkksTjmdRalm6baj8aI","remoteAddr":"***Censored***","app":"DeleteOrphanedSharesJob","message":"0 orphaned share(s) deleted","level":0,"time":"2017-03-31T07:33:00+00:00","method":"GET","url":"\/cron.php","user":"--"}
{"reqId":"dsQdTS2GxCWdbr9F3EYB","remoteAddr":"***Censored***","app":"no app in context","message":"Current user is not logged in","level":0,"time":"2017-03-31T07:40:20+00:00","method":"GET","url":"\/index.php\/apps\/files\/","user":"--"}
```

#### Browser log
This was captured from a later try with a different user, but with the same results. Entry from the network log:
```
POST XHR http://93.190.76.104/index.php/settings/users/changepassword [HTTP/1.1 503 Service Unavailable 332ms]
Raw Post Data: username=kevin2&password=***new user password***&recoveryPassword=***recovery key***

Response (body only):
<body id="body-login">
		<noscript>
	<div id="nojavascript">
		<div>
			This application requires JavaScript for correct operation. Please <a href="http://enable-javascript.com/" target="_blank" rel="noreferrer">enable JavaScript</a> and reload the page.		</div>
	</div>
</noscript>
		<div class="wrapper">
			<div class="v-align">
									<header role="banner">
						<div id="header">
							<div class="logo svg">
								<h1 class="hidden-visually">
									ownCloud								</h1>
							</div>
							<div id="logo-claim" style="display:none;"></div>
						</div>
					</header>
								<ul class="error-wide">
			<li class='error'>
			Private Key missing for user: please try to log-out and log-in again<br>
					</li>
	</ul>
				<div class="push"></div><!-- for sticky footer -->
			</div>
		</div>
		<footer role="contentinfo">
			<p class="info">
				<a href="https://owncloud.org" target="_blank" rel="noreferrer">ownCloud</a> – web services under your control			</p>
		</footer>
	</body>

```

**Comment by PVince81** [comment]
@sharidas can you check whether you can reproduce this ? Looks suspicious

**Comment by eliawry** [comment]
We are having the exact same issue, and the user has lost access to all of their files. We have file backups, but we have been forced to just create a new user.

**Comment by tervor** [comment]
@sharidas, where you able to reproduce this?

**Comment by sharidas** [comment]
@tervor If I understand the steps to reproduce, there are 2 users. Below are the steps which I have followed, let me know if this is what you have done, please:

1. Created an admin user ( which belongs to admin group )
2. Enabled Default encryption app
3. In the Admin->Server-side encryption check "Enable server-side encryption" and click "Enable encryption" confirmation button.
4. Logout and login back to the admin user so that you can set the recovery key.
5. Enable the recovery key.
6. Create a user, user1
7. login as user1
8. Select the radio button "Enabled" under Personal - > ownCloud basic encryption module -> Enable password recovery . Create a text file in user1 home. And add some data to it.
9. Login back to admin user
10. Reset user1 password.
11. Also reset the recovery key password.
12. Login back to the user1.
13. I am able to read the data of the text file from the browser. The text file is encrypted.

May I know how did you verified that user keys were not updated? Was there any issue while you were trying to open a file or so? If so can you share the same?

**Comment by sharidas** [comment]
I gave a try with 9.1.5 version. Unfortunately I am not able to reproduce. @tervor Let me know if I misunderstood your steps?

**Comment by tervor** [comment]
@sharidas, I didn't do the step 11. I just have reset the users password as in step 10 with the recovery key filled in into the recovery key field (on top of the users list) in the web interface.
I can retest this with 9.1.5. The original request was for 9.1.4.

**Comment by sharidas** [comment]
@tervor Sure let me know with 9.1.5. You can ignore the step 11 mentioned in https://github.com/owncloud/core/issues/27547#issuecomment-296713193.

**Comment by tervor** [comment]
@sharidas,
With 9.1.5 it seems to work as expected. I also did a test with a fresh 9.1.4 installation. There it also works as expected. The previous test installation was updated from 9.1.0 to 9.1.4 previously. Maybe this was causing the issue.
Previously I got no confirmation after resetting a users password. In the two fresh test installations, I got a `Password successfully changed` message on top of the web interface.

Maybe to fully answer the questions in [your comment above](https://github.com/owncloud/core/issues/27547#issuecomment-296713193): 

> May I know how did you verified that user keys were not updated? Was there any issue while you were trying to open a file or so? If so can you share the same?

I got a warning when i logged in with the user on top of the page: 

    Invalid private key for Encryption App. Please update your private key password in your personal settings to recover access to your encrypted files.

If I tried to access a file in the users account I got this error:

    An error occurred!

    Private Key missing for user: please try to log-out and log-in again
